sdk/tables/azure-tables/review/tables.api.md

```diff
@@ -455,10 +455,18 @@ export interface TableQueryHeaders {
 // @public
 export interface TableQueryOptionalParams extends coreHttp.OperationOptions {
     nextTableName?: string;
-    queryOptions?: QueryOptions;
+    queryOptions?: TableQueryOptions;
     requestId?: string;
 }
 
+// @public
+export interface TableQueryOptions {
```

Comment:
Might be a personal issue that I just need to deal with, but I don't really love having this exposed publicly now since I don't think we end up using it in practice?  Like we would never expect someone to interact with this type directly, right?  It's public so that we can do the `Omit<TableQueryOptions, "select">` thing?

Reply:
I'll replace Omit Types with new re-defined interfaces next to address #10534. With this we'll be able to stop exporting un-used generated types